### PR TITLE
fix: check duplicates while adding dependencies

### LINKED ISSUE #825: Don't leave `pixi.toml` in broken state
### Checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://github.com/prefix-dev/pixi/releases) of pixi, using `pixi --version`.

### Reproducible example

```shell
pixi add flask
pixi add Flask
pixi install
```

### Issue description

The before mentioned example results in an error on the `install` because we don't allow duplicates. 
But `add` allows you to add duplicates

### Expected behavior

Don't allow duplicates. Error on the add if the normalized version already exists in toml

### PR DESCRIPTION
**Pull request by orhun**
closes #825

### CODE REVIEW

File: src/cli/add.rs [hunk 1]
```diff
@@ -217,9 +217,23 @@ pub async fn add_pypi_specs_to_project(
...
             project.manifest.add_pypi_dependency(name, spec, None)?;
         } else {
             for platform in specs_platforms.iter() {
+                let dependencies = project.pypi_dependencies(Some(*platform));
+                if let Some(package_name) = dependencies.keys().find(|pkg_name| &name == pkg_name) {
+                    return Err(miette::miette!(
+                        "{} is already added as dependency.",
+                        console::style(package_name.as_source_str()).bold(),
+                    ));
+                }
```
Comment (ruben-arts) [inline src/cli/add.rs:236]: I guess you could add these checks to `add_(pypi_)dependency`

Comment (orhun) [reply src/cli/add.rs:236]: I can't access the `dependencies` from that context (`Manifest`). Is there a way of doing that? I'm guessing passing `dependencies` as argument is not optimal.

Comment (orhun) [reply src/cli/add.rs:236]: nvm I think I found a way

File: src/project/manifest/mod.rs [hunk 2]
```diff
@@ -355,6 +355,14 @@ impl Manifest {
             miette::bail!("pixi does not support wildcard dependencies")
         };
 
+        // Check for duplicates.
+        if dependency_table.contains_key(name.as_source()) {
```
Comment (ruben-arts) [inline src/project/manifest/mod.rs:359]: You want the normalized version

**Comment by ruben-arts** [comment]
It think it is fine if we don't allow `pixi add` to add capitalization. 

Btw, it doesn't work yet. Im testing with:
```
pixi add flask
pixi add Flask # this should already fail
pixi install # but it fails here
```

File: src/project/manifest/mod.rs (hunk 2 above)
Comment (orhun) [reply src/project/manifest/mod.rs:359]: It is inserted into the table as source:

```rs
dependency_table.insert(name.as_source(), Item::Value(spec.to_string().into()));
```

Comment (ruben-arts) [reply src/project/manifest/mod.rs:359]: Yeah so either insert as normalized or make the check normalized. I would say insert as normalized. Not sure why you want to add it as source

**Review by ruben-arts — APPROVED** [review]

**Pull request merged**